odoo/addons/base/models/ir_actions_report.py

```diff
@@ -730,7 +743,7 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
             pdf_content_stream = io.BytesIO(pdf_content)
 
             # Printing a PDF report without any records. The content could be returned directly.
-            if not res_ids:
+            if has_duplicated_ids or not res_ids:
```

Comment:
Useless ? has_duplicated_ids cannot be true it there isn't res_ids.

Reply:
In case of `has_duplicated_ids`, we want to completely skip the splitting of the PDF to create the attachment (the `report_sudo.attachment` feature). So in that case, I do the same as when there is no res_ids, I just return the stream containing the full content without indexing by res_id